src/main/java/org/apache/tools/bzip2/CBZip2InputStream.java

```diff
@@ -29,957 +29,918 @@
-		final Data dataShadow = this.data;
-		final char[][] len = dataShadow.temp_charArray2d;
-		final int[] minLens = dataShadow.minLens;
-		final int[][] limit = dataShadow.limit;
-		final int[][] base = dataShadow.base;
-		final int[][] perm = dataShadow.perm;
-		
-		for (int t = 0; t < nGroups; t++) {
-			int minLen = 32;
-			int maxLen = 0;
-			final char[] len_t = len[t];
-			for (int i = alphaSize; --i >= 0;) {
-				final char lent = len_t[i];
-				if (lent > maxLen) {
-					maxLen = lent;
-				}
-				if (lent < minLen) {
-					minLen = lent;
-				}
-			}
-			hbCreateDecodeTables(limit[t], base[t], perm[t], len[t], minLen, maxLen, alphaSize);
-			minLens[t] = minLen;
-		}
-	}
-	
-	private void getAndMoveToFrontDecode() throws IOException {
-		this.origPtr = bsR(24);
-		recvDecodingTables();
-		
-		final InputStream inShadow = this.in;
-		final Data dataShadow = this.data;
-		final byte[] ll8 = dataShadow.ll8;
-		final int[] unzftab = dataShadow.unzftab;
-		final byte[] selector = dataShadow.selector;
-		final byte[] seqToUnseq = dataShadow.seqToUnseq;
-		final char[] yy = dataShadow.getAndMoveToFrontDecode_yy;
-		final int[] minLens = dataShadow.minLens;
-		final int[][] limit = dataShadow.limit;
-		final int[][] base = dataShadow.base;
-		final int[][] perm = dataShadow.perm;
-		final int limitLast = this.blockSize100k * 100000;
+        for (int t = 0; t < nGroups; t++) {
+            int curr = bsR(5);
+            final char[] len_t = len[t];
+            for (int i = 0; i < alphaSize; i++) {
+                while (bsGetBit()) {
+                    curr += bsGetBit() ? -1 : 1;
+                }
+                len_t[i] = (char) curr;
+            }
+        }
+
+        // finally create the Huffman tables
+        createHuffmanDecodingTables(alphaSize, nGroups);
+    }
+
+    /**
+     * Called by recvDecodingTables() exclusively.
+     */
+    private void createHuffmanDecodingTables(final int alphaSize, final int nGroups) {
+        final Data dataShadow = this.data;
+        final char[][] len = dataShadow.temp_charArray2d;
+        final int[] minLens = dataShadow.minLens;
+        final int[][] limit = dataShadow.limit;
+        final int[][] base = dataShadow.base;
+        final int[][] perm = dataShadow.perm;
+
+        for (int t = 0; t < nGroups; t++) {
+            int minLen = 32;
+            int maxLen = 0;
+            final char[] len_t = len[t];
+            for (int i = alphaSize; --i >= 0; ) {
+                final char lent = len_t[i];
+                if (lent > maxLen) {
+                    maxLen = lent;
+                }
+                if (lent < minLen) {
+                    minLen = lent;
+                }
+            }
+            hbCreateDecodeTables(limit[t], base[t], perm[t], len[t], minLen, maxLen, alphaSize);
+            minLens[t] = minLen;
+        }
+    }
+
+    private void getAndMoveToFrontDecode() throws IOException {
+        this.origPtr = bsR(24);
+        recvDecodingTables();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://titan.maxgamer.org:9000/coding_rules#rule_key=squid%3AS00117)